backend/tests/test_exceptions.py

```diff
@@ -0,0 +1,155 @@
+from fastapi.testclient import TestClient
+from sqlalchemy.exc import NoResultFound
+import pytest
+from rope.api.main import app
+from rope.api.sessions import get_request_session
+from rope.api.database import SessionLocal
+from rope.db.schema import UserAccount
+
+
+@pytest.fixture
+def test_client():
+    return TestClient(app)
+
+
+@pytest.fixture
+def db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@pytest.fixture(autouse=True)
+def clear_database_table(db):
+    db.query(UserAccount).delete()
+    db.commit()
+
+
+@pytest.fixture
+def setup_admin_session(mocker):
```

Comment:
To your question [here](https://github.com/openstax/rope/pull/5#issuecomment-1932726331) related to shared fixtures: It's sort of a judgement call. I think some of the fixtures like `db()` and `test_client()` might be nice to have shared to DRY things up. Maybe some that are going to impact the database like `clear_database_table` are better to include in files since not all tests might need that + it is `autouse=True` so it limits scope.